src/System.Text.Json/tests/JsonElementWriteTests.cs

```diff
@@ -460,6 +464,122 @@ public static void WriteNumberScientificAsProperty(bool indented)
                 "{\"m\\u00eal\\u00e9e\":1e6}");
         }
 
+        [Fact]
+        public static void WriteValueUnicodeEscapeSequence()
+        {
+            string unicodeString = "\u0421\u0430\u0439\u043D \u0443\u0443? \u04E9\u04E9";
+            string json = $"[\"{unicodeString}\"]";
+            var buffer = new ArrayBufferWriter<byte>(1024);
+            ReadOnlySequence<byte> sequence = JsonTestHelper.GetSequence(Encoding.UTF8.GetBytes(json), 1);
+            string expectedStr = GetEscapedExpectedString(unicodeString, StringEscapeHandling.EscapeNonAscii);
+
+            using (JsonDocument doc = JsonDocument.Parse(sequence, s_options))
```

Comment:
Many of these tests seem like they'd be more appropriate at the writer level, because they're not using anything specific from JsonDocument.  Am I missing JsonDocument's involvement?

Reply:
Not sure what you mean exactly. It is getting JsonElement from JsonDocument, the test is for JsonElement.WriteTo not for Utf8JsonWriter

Reply:
I understand that the test _is_ using JsonDocument/JsonElement, but the question is "why?"  What's the scenario that's being tested that requires using JsonDocument/JsonElement.  The method name feels like it's testing something in the writer, which means that the test could be written without using JsonDocument/JsonElement at all.

It's using sequences, which aren't really interesting to JsonDocument, because JsonDocument.Parse just drains all the sequence contents into one rented array; so the write isn't doing anything clever.

I'm really just trying to understand what this test is testing... what were we not already testing that we are now?

Reply:
That might explain why I was not able to hit error nous conditions wanted to cover with whatever bad data used...

I just trying to improve coverage for JsonWriterHelper.Escaping.cs, I added this because it was calling https://github.com/dotnet/corefx/blob/master/src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Escaping.cs#L234 for which I was adding coverage. But I see that reference is removed just 3 hours ago …

As you know I don't really familiar with underlying code design just following what is calling the those section or what tests running around that. But seems the code i was trying to cove was invalid.